source/common/mongo/proxy.cc

```diff
@@ -241,5 +245,54 @@ DecoderPtr ProdProxyFilter::createDecoder(DecoderCallbacks& callbacks) {
   return DecoderPtr{new DecoderImpl(callbacks)};
 }
 
+FaultConfigImpl::FaultConfigImpl(const Json::Object& fault_config) {
+  const Json::ObjectSharedPtr delay_config = fault_config.getObject("delay");
+
+  delay_percent_ = static_cast<uint32_t>(delay_config->getInteger("percent"));
+  duration_ms_ = static_cast<uint64_t>(delay_config->getInteger("duration_ms"));
+}
+
+Optional<uint64_t> ProxyFilter::delayDuration() {
+  Optional<uint64_t> result;
+
+  if (!fault_config_)
+    return result;
+
+  if (!runtime_.snapshot().featureEnabled("mongo.fault.delay.percent",
+                                          fault_config_->delayPercent())) {
+    return result;
+  }
+
+  uint64_t duration = runtime_.snapshot().getInteger("mongo.fault.delay.duration_ms",
+                                                     fault_config_->delayDuration());
+
+  // Delay only if the duration is > 0ms.
+  if (duration > 0) {
+    result.value(duration);
```

Comment:
It makes sense to have faults which are distribution driven as well.
This PR is for the simple case to get a parity with HTTP faults filter and kicking off initial testing. Current implementation injecting fixed latency for some % of requests.
I can name it explicitly fixed_delay to avoid confusions as well.